RVV branch is broken in new DNN engine

### System Information

OS: Linux
Platform: RISC-V with RVV.

List of failed tests without models downloading:
```
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_1d_default_OCV_CPU, where GetParam() = (test_averagepool_1d_default, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_ceil_OCV_CPU, where GetParam() = (test_averagepool_2d_ceil, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_default_OCV_CPU, where GetParam() = (test_averagepool_2d_default, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_pads_OCV_CPU, where GetParam() = (test_averagepool_2d_pads, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_precomputed_pads_OCV_CPU, where GetParam() = (test_averagepool_2d_precomputed_pads, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_precomputed_same_upper_OCV_CPU, where GetParam() = (test_averagepool_2d_precomputed_same_upper, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_precomputed_strides_OCV_CPU, where GetParam() = (test_averagepool_2d_precomputed_strides, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_same_upper_OCV_CPU, where GetParam() = (test_averagepool_2d_same_upper, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_2d_strides_OCV_CPU, where GetParam() = (test_averagepool_2d_strides, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_averagepool_3d_default_OCV_CPU, where GetParam() = (test_averagepool_3d_default, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_basic_conv_with_padding_OCV_CPU, where GetParam() = (test_basic_conv_with_padding, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_basic_conv_without_padding_OCV_CPU, where GetParam() = (test_basic_conv_without_padding, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_conv_with_autopad_same_OCV_CPU, where GetParam() = (test_conv_with_autopad_same, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_conv_with_strides_and_asymmetric_padding_OCV_CPU, where GetParam() = (test_conv_with_strides_and_asymmetric_padding, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_conv_with_strides_no_padding_OCV_CPU, where GetParam() = (test_conv_with_strides_no_padding, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_conv_with_strides_padding_OCV_CPU, where GetParam() = (test_conv_with_strides_padding, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_1d_default_OCV_CPU, where GetParam() = (test_maxpool_1d_default, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_ceil_OCV_CPU, where GetParam() = (test_maxpool_2d_ceil, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_default_OCV_CPU, where GetParam() = (test_maxpool_2d_default, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_pads_OCV_CPU, where GetParam() = (test_maxpool_2d_pads, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_precomputed_pads_OCV_CPU, where GetParam() = (test_maxpool_2d_precomputed_pads, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_precomputed_same_upper_OCV_CPU, where GetParam() = (test_maxpool_2d_precomputed_same_upper, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_precomputed_strides_OCV_CPU, where GetParam() = (test_maxpool_2d_precomputed_strides, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_same_upper_OCV_CPU, where GetParam() = (test_maxpool_2d_same_upper, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_2d_strides_OCV_CPU, where GetParam() = (test_maxpool_2d_strides, OCV/CPU)
[  FAILED  ] Test_ONNX_conformance.Layer_Test/test_maxpool_3d_default_OCV_CPU, where GetParam() = (test_maxpool_3d_default, OCV/CPU)
[  FAILED  ] Test_ONNX_layers.MaxPooling/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.MaxPooling_2/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.Convolution_variable_weight_bias/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.Convolution3D_bias/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.MaxPooling_Sigmoid/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.AveragePooling/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.MaxPooling3D/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.AvePooling3D/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.PoolConv3D/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.Constant/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.CalculatePads/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.Conv1d_variable_weight_bias/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.DynamicAxes_average_pooling/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.DynamicAxes_maxpooling_sigmoid/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.DynamicAxes_dynamic_batch/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.MaxPool1d/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.MaxPoolSigmoid1d/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.MaxPool1d_Twise/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.AvePool1d/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.PoolConv1d/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.ConvResizePool1d/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.DepthWiseAdd/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_layers.DepthStride2/0, where GetParam() = OCV/CPU
[  FAILED  ] Test_ONNX_nets.MobileNet_v4/0, where GetParam() = OCV/CPU
```

### Detailed description

Test examples:
```
[ RUN      ] Test_ONNX_layers.MaxPooling/0, where GetParam() = OCV/CPU
[ WARN:0@0.147] global net_impl_backend.cpp:334 setPreferableTarget Targets are not supported by the new graph engine for now
unknown file: Failure
C++ exception with description "OpenCV(5.0.0-pre) /mnt/Projects/Projects/opencv-next/modules/dnn/src/layers/maxpool_layer.cpp:56: error: (-215:Assertion failed) C0 == nlanes || C0 == nlanes*2 || C0 % (nlanes*4) == 0 in function 'operator()'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.MaxPooling/0, where GetParam() = OCV/CPU (6 ms)
[----------] 1 test from Test_ONNX_layers (6 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (7 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test_ONNX_layers.MaxPooling/0, where GetParam() = OCV/CPU
```
```
[ RUN      ] Test_ONNX_layers.Convolution_variable_weight_bias/0, where GetParam() = OCV/CPU
[ WARN:0@0.148] global net_impl_backend.cpp:334 setPreferableTarget Targets are not supported by the new graph engine for now
unknown file: Failure
C++ exception with description "OpenCV(5.0.0-pre) /mnt/Projects/Projects/opencv-next/modules/dnn/src/layers/cpu_kernels/conv2_depthwise.simd.hpp:96: error: (-215:Assertion failed) C0 == nlanes || C0 == nlanes*2 || C0 % (nlanes*4) == 0 in function 'operator()'
" thrown in the test body.
[  FAILED  ] Test_ONNX_layers.Convolution_variable_weight_bias/0, where GetParam() = OCV/CPU (11 ms)
[----------] 1 test from Test_ONNX_layers (11 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (12 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Test_ONNX_layers.Convolution_variable_weight_bias/0, where GetParam() = OCV/CPU
```


### Steps to reproduce

Build commands (cross):
```
make -DCMAKE_C_COMPILER=$SDK_DIR/bin/riscv64-unknown-linux-gnu-gcc       -DCMAKE_CXX_COMPILER=$SDK_DIR/bin/riscv64-unknown-linux-gnu-g++       -DCMAKE_TOOLCHAIN_FILE=../opencv/platforms/linux/riscv64-gcc.toolchain.cmake       -DCPU_BASELINE=RVV -DCPU_BASELINE_REQUIRE=RVV -DRISCV_RVV_SCALABLE=ON -DWITH_OPENCL=OFF ../opencv-next && make -j12
```

Test command:
```
./opencv_test_dnn
```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/dnn/src/layers/avgpool_layer.cpp
+++ b/modules/dnn/src/layers/avgpool_layer.cpp
@@ -52,7 +52,7 @@ static void avgPool32f(const void* inp_, void* out_,
         float* out = (float*)out_ + nc0*planesize;
         float iksize = 1.f/ksize;
 
-#if CV_SIMD || CV_SIMD_SCALABLE
+#if CV_SIMD
         int nlanes = VTraits<v_float32>::vlanes();
         CV_Assert(C0 == nlanes || C0 == nlanes*2 || C0 % (nlanes*4) == 0);
         v_float32 z = vx_setzero_f32();
@@ -68,12 +68,12 @@ static void avgPool32f(const void* inp_, void* out_,
                         y0 >= inner_y0 && y0 < inner_y1 ? inner_x0 : W;
                     int yi_ = y0*SY - padY0;
 
-                #if !(CV_SIMD || CV_SIMD_SCALABLE)
+                #if !(CV_SIMD)
                     memset(out, 0, W*C0*sizeof(out[0]));
                 #endif
 
                     for(;;) {
-                    #if CV_SIMD || CV_SIMD_SCALABLE
+                    #if CV_SIMD
                         if (nlanes == C0) {
                             for (; x0 < x1; x0++) {
                                 int xi_ = x0*SX - padX0;
@@ -152,7 +152,7 @@ static void avgPool32f(const void* inp_, void* out_,
                             break;
                         x1 = inner_x1;
 
-                    #if CV_SIMD || CV_SIMD_SCALABLE
+                    #if CV_SIMD
                         if (nlanes == C0) {
                             for (; x0 < x1; x0++) {
                                 int xi_ = x0*SX - padX0;

--- a/modules/dnn/src/layers/batch_norm2_layer.cpp
+++ b/modules/dnn/src/layers/batch_norm2_layer.cpp
@@ -19,7 +19,7 @@ namespace dnn {
 */
 
 #undef CV_SIMD_ONLY
-#if CV_SIMD || CV_SIMD_SCALABLE
+#if CV_SIMD
 #define CV_SIMD_ONLY(expr) expr
 #else
 #define CV_SIMD_ONLY(expr)
@@ -135,7 +135,7 @@ static void batchnorm(const Mat& inp, Mat& out, const Mat& scale,
                     }
                 }
             }
-        #if CV_SIMD || CV_SIMD_SCALABLE
+        #if CV_SIMD
             /*
                 [TODO] support C0 == vlanes/2, maybe C0 == vlanes/4.
                 in this case, load everything into vsc0 and vb0, process

--- a/modules/dnn/src/layers/cpu_kernels/conv2_deconv.cpp
+++ b/modules/dnn/src/layers/cpu_kernels/conv2_deconv.cpp
@@ -63,7 +63,7 @@ static void deconvBlock32f(const void* inp__, const void* /*residual*/,
     const float* wdata = (const float*)weights__;
     const float* bias  = bias__;
 
-#if (CV_SIMD || CV_SIMD_SCALABLE)
+#if (CV_SIMD)
     // SIMD path is safe when ngroups==1 or Kg%C0==0; repackDeconvWeights()
     // zero-fills padded lanes.
     const bool simd_ok =
@@ -83,7 +83,7 @@ static void deconvBlock32f(const void* inp__, const void* /*residual*/,
             float* out_k1 = (float*)out__ + ((int64_t)n * K1 + k1) * ospatial * C0;
             const float* inp_n = (const float*)inp__ + (int64_t)n * C1 * ispatial * C0;
 
-#if (CV_SIMD || CV_SIMD_SCALABLE)
+#if (CV_SIMD)
             // Precompute the shared (g, kblk, c1_abs_base) for the SIMD path.
             int simd_g = 0, simd_kblk = 0, simd_c1_abs_base = 0;
             if (simd_ok) {
@@ -115,7 +115,7 @@ static void deconvBlock32f(const void* inp__, const void* /*residual*/,
                     }
                 }
 
-#if (CV_SIMD || CV_SIMD_SCALABLE)
+#if (CV_SIMD)
                 if (simd_ok) {
                     const int VLANES = VTraits<v_float32>::vlanes();
                     const int v_per_block = C0 / VLANES;

--- a/modules/dnn/src/layers/cpu_kernels/conv2_depthwise.simd.hpp
+++ b/modules/dnn/src/layers/cpu_kernels/conv2_depthwise.simd.hpp
@@ -89,7 +89,7 @@ static void depthwiseConv32f(const void* inp__, const void* residual__,
             defaultAlpha = 1.f;
         }
 
-    #if CV_SIMD || CV_SIMD_SCALABLE
+    #if CV_SIMD
         v_float32 v_maxval = vx_setall_f32(maxval);
         v_float32 z = vx_setzero_f32();
         const int nlanes = VTraits<v_float32>::vlanes();
@@ -126,12 +126,12 @@ static void depthwiseConv32f(const void* inp__, const void* residual__,
                         y0 >= inner_y0 && y0 < inner_y1 ? inner_x0 : W;
                     int yi_ = y0*SY - padY0;
 
-                #if !(CV_SIMD || CV_SIMD_SCALABLE)
+                #if !(CV_SIMD)
                     memset(out, 0, W*C0*sizeof(out[0]));
                 #endif
 
                     for(;;) {
-                    #if CV_SIMD || CV_SIMD_SCALABLE
+                    #if CV_SIMD
                         if (nlanes == C0) {
                             v_float32 sc0 = vx_load(scalebuf), b0 = vx_load(biasbuf);
                             v_float32 alpha0 = vx_load(alphabuf);
@@ -218,7 +218,7 @@ static void depthwiseConv32f(const void* inp__, const void* residual__,
                             break;
                         x1 = inner_x1;
 
-                    #if CV_SIMD || CV_SIMD_SCALABLE
+                    #if CV_SIMD
                         if (nlanes == C0) {
                             v_float32 sc0 = vx_load(scalebuf), b0 = vx_load(biasbuf), alpha0 = vx_load(alphabuf);
                             for (; x0 < x1; x0++) {
@@ -335,7 +335,7 @@ static void depthwiseConv32f(const void* inp__, const void* residual__,
                         x1 = W;
                     }
 
-                #if !(CV_SIMD || CV_SIMD_SCALABLE)
+                #if !(CV_SIMD)
                     if (residual) {
                         for (int x = 0; x < W*C0; x += C0) {
                             for (int c = 0; c < C0; c++) {

--- a/modules/dnn/src/layers/maxpool_layer.cpp
+++ b/modules/dnn/src/layers/maxpool_layer.cpp
@@ -50,7 +50,7 @@ static void maxPool32f(const void* inp_, void* out_, const ConvState& cs)
         float* out = (float*)out_ + nc0*planesize;
         const float INITVAL = -FLT_MAX;
 
-    #if CV_SIMD || CV_SIMD_SCALABLE
+    #if CV_SIMD
         int nlanes = VTraits<v_float32>::vlanes();
         v_float32 s_min = vx_setall_f32(INITVAL);
         CV_Assert(C0 == nlanes || C0 == nlanes*2 || C0 % (nlanes*4) == 0);
@@ -65,13 +65,13 @@ static void maxPool32f(const void* inp_, void* out_, const ConvState& cs)
                         y0 >= inner_y0 && y0 < inner_y1 ? inner_x0 : W;
                     int yi_ = y0*SY - padY0;
 
-                #if !(CV_SIMD || CV_SIMD_SCALABLE)
+                #if !(CV_SIMD)
                     for (int c = 0; c < C0*W; c++)
                         out[c] = INITVAL;
                 #endif
 
                     for(;;) {
-                    #if CV_SIMD || CV_SIMD_SCALABLE
+                    #if CV_SIMD
                         if (nlanes == C0) {
                             for (; x0 < x1; x0++) {
                                 int xi_ = x0*SX - padX0;
@@ -136,7 +136,7 @@ static void maxPool32f(const void* inp_, void* out_, const ConvState& cs)
                             break;
                         x1 = inner_x1;
 
-                    #if CV_SIMD || CV_SIMD_SCALABLE
+                    #if CV_SIMD
                         if (nlanes == C0) {
                             for (; x0 < x1; x0++) {
                                 int xi_ = x0*SX - padX0;

